### PR TITLE
Firefox support

### PR DESCRIPTION
**Pull request by TheHandsomeCoder**

### CODE REVIEW

File: app/scripts/dimApp.config.js [hunk 1]
```diff
@@ -77,12 +76,13 @@
         $window.initgapi = function() {
           SyncService.init();
         }
+
         SyncService.get().then(function(data) {
-          data = (data) ? data['2016.03.13-v3.4.0'] : null;          
-          if(_.isNull(data) || _.isEmpty(data)) {
+          var toastViewedFlag = data['2016.03.13-v3.4.0'] || {};
+          if(_.isNull(toastViewedFlag)) {
```
Comment (bhollis) [inline app/scripts/dimApp.config.js]: With your change, will `toastViewedFlag` ever be null?

File: app/scripts/loadout/dimLoadout.directive.js [hunk 2]
```diff
@@ -96,9 +96,9 @@
     }
   }
 
-  LoadoutCtrl.$inject = ['dimLoadoutService', 'dimCategory', 'dimItemTier', 'toaster', 'dimPlatformService', 'dimSettingsService', '$scope'];
+  LoadoutCtrl.$inject = ['dimLoadoutService', 'dimCategory', 'dimItemTier', 'toaster', 'dimPlatformService', 'dimSettingsService', '$scope','dimStoreService'];
 
-  function LoadoutCtrl(dimLoadoutService, dimCategory, dimItemTier, toaster, dimPlatformService, dimSettingsService, $scope) {
+  function LoadoutCtrl(dimLoadoutService, dimCategory, dimItemTier, toaster, dimPlatformService, dimSettingsService, $scope, dimStoreService) {
```
Comment (bhollis) [inline app/scripts/loadout/dimLoadout.directive.js]: Is this used anywhere?

File: app/scripts/services/dimBungieService.factory.js [hunk 3]
```diff
@@ -101,7 +102,7 @@
               });
 
               reject(new Error('No bungled cookie found.'));
-            }
+            }
```
Comment (bhollis) [inline app/scripts/services/dimBungieService.factory.js]: Would you mind reverting these style changes?

File: app/scripts/services/dimBungieService.factory.js [hunk 4]
```diff
@@ -143,24 +144,27 @@
...
-              active: false
-            });
-          }
-        });
-
+        try {
+          //TODO: This is listed as supported on the firefox developer docs but throws a malformed URI error
+          chrome.tabs.query({
+            'url': '*://*.bungie.net/'
+          }, function(tabs) {
```
Comment (bhollis) [inline app/scripts/services/dimBungieService.factory.js]: We actually can't do this from chrome anymore either - we don't ask for the permission. If you rebase it should be gone.

File: app/scripts/services/dimBungieService.factory.js [hunk 5]
```diff
@@ -143,24 +144,27 @@
...
+            if (_.size(tabs) === 0) {
+              chrome.tabs.create({
+                url: 'https://www.bungie.net/',
+                active: false
+              });
+            }
+          })
+        } catch (err) {          
+          console.log(err);
```
Comment (bhollis) [inline app/scripts/services/dimBungieService.factory.js]: `console.error`?

**Comment by bhollis** [comment]
@TheHandsomeCoder this is fantastic! One thing that'd really help with getting this change merged in, though, is if you could separate out the "make it work on Firefox" bits from style changes and migrating to Sass, adding Grunt stuff, etc. I like those changes too, but I'd like them even better if I could merge them as independent PRs. :stuck_out_tongue:

File: app/scripts/dimApp.config.js (hunk 1 above)
Comment (TheHandsomeCoder) [inline app/scripts/dimApp.config.js]: This was null originally. I think I changed it back as part of some smoke tests. I've reverted it to null and wrapped the entire block in a null check for the data variable as it was throwing console errors

File: app/scripts/loadout/dimLoadout.directive.js (hunk 2 above)
Comment (TheHandsomeCoder) [inline app/scripts/loadout/dimLoadout.directive.js]: Being used on line 253. It was throwing an undefined error in FF which was preventing DIM loading

File: app/scripts/services/dimBungieService.factory.js (hunk 3 above)
Comment (TheHandsomeCoder) [inline app/scripts/services/dimBungieService.factory.js]: Sure thing :+1:

**Comment by bhollis** [comment]
Merged into `sync-svc`.

**Pull request merged**